code-challenges.test.js

```diff
@@ -13,36 +13,102 @@
+        // Example input: 6
+        // Expected output: [1, 1, 2, 3, 5, 8]
+        
+        // Example input: 10
+        // Expected output: [1, 1, 2, 3, 5, 8, 13, 21, 34, 55]
+  
+      //an expect method, nested within the test block, calling on the hello() function, followed by the .toEqual() matcher that checks the expected output of the function return.
+      expect(fibonacci(6)).toEqual([1, 1, 2, 3, 5, 8])
+      expect(fibonacci(10)).toEqual([1, 1, 2, 3, 5, 8, 13, 21, 34, 55])
+    })
+  })
 
-// Example input: 10
-// Expected output: [1, 1, 2, 3, 5, 8, 13, 21, 34, 55]
 
 
 // b) Create the function that makes the test pass.
 
+const fibonacci = (arraySize) => {
+    const fibArr = [1, 1]
+
+    for (let index = 0; index < arraySize - 2; index++) {
+        fibArr.push((fibArr[index] + fibArr[index + 1]))        
+    }
+    return fibArr
+}
+
+
+// console.log(fibonacci(5));
 
 // --------------------2) Create a function that takes in an array and returns a new array of only odd numbers sorted from least to greatest.
+// create function using .filter, and method typeOf
+// if value at index i; typeOf === "number" push into a new Arr
+// create another function that takes in the returned array, and checks to see if number is odd
+// if arr[i] % 2 === 1 || arr[i] % 2 === -1 return it to the new array
+// return that array with .sort() function
+// sort function only sorts using first index of whatever you are looking at 30 > 400000 because 3 > 4
+// need to use .sort(function(a, b){return a-b})
 
 // a) Create a test with expect statements for each of the variables provided.
 
-var fullArr1 = [4, 9, 0, "7", 8, true, "hey", 7, 199, -9, false, "hola"]
-// Expected output: [-9, 7, 9, 199]
+describe("arrFiltSort", () => {
+
+    // a test/it method, nested within the describe block, that in plain words, describes that the function does.
+    it("Takes in a number that represents the length of an array, and returns an array following the fibonacci sequence of that array size", () => {
+
+        var fullArr1 = [4, 9, 0, "7", 8, true, "hey", 7, 199, -9, false, "hola"]
+        // Expected output: [-9, 7, 9, 199]
+        
+        var fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
+        // Expected output: [-823, 7, 23]
+  
+      //an expect method, nested within the test block, calling on the hello() function, followed by the .toEqual() matcher that checks the expected output of the function return.
+      expect(arrFiltSort(fullArr1)).toEqual([-9, 7, 9, 199])
+      expect(arrFiltSort(fullArr2)).toEqual([-823, 7, 23])
+    })
+  })
 
-var fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
-// Expected output: [-823, 7, 23]
 
 
 // b) Create the function that makes the test pass.
 
+const arrFiltSort = (array) => {
+    return array.filter(value => (value % 2 === 1 || value % 2 === -1) && typeof value === "number").sort(function(a,b){return a-b})
+}
+
```

Comment:
Beautiful! One refactor could be using the arrow syntax in the sort method. That would allow you to remove the curlies and return inside that method.